src/lnurl.js

```diff
@@ -0,0 +1,116 @@
+import bech32 from 'bech32'
+import wrap from './lib/promise-wrap'
+
+const debug = require('debug')('lightning-charge')
+
+module.exports = (app, payListen, model, auth) => {
+  const {
+    newInvoice, listInvoicesByLnurlPayEndpoint
+  , getLnurlPayEndpoint, listLnurlPayEndpoints
+  , setLnurlPayEndpoint, delLnurlPayEndpoint
+  } = model
+
+  app.get('/endpoints', auth, wrap(async (req, res) =>
+    res.status(200).send(
+      (await listLnurlPayEndpoints())
+        .map(lnurlpay => addBech23Lnurl(req, lnurlpay))
+    )))
+
+  app.post('/endpoint', auth, wrap(async (req, res) =>
+    res.status(201).send(
+      addBech23Lnurl(req, await setLnurlPayEndpoint(null, req.body))
+    )))
+
+  app.put('/endpoint/:id', auth, wrap(async (req, res) =>
+    res.status(200).send(
+      addBech23Lnurl(req, await setLnurlPayEndpoint(req.params.id, req.body))
+    )))
+
+  app.delete('/endpoint/:id', auth, wrap(async (req, res) =>
+    res.status(200).send(await delLnurlPayEndpoint(req.params.id))))
+
+  app.get('/endpoint/:id', auth, wrap(async (req, res) =>
+    res.status(200).send(
+      addBech23Lnurl(req, await getLnurlPayEndpoint(req.params.id))
+    )))
+
+  app.get('/endpoint/:id/invoices', auth, wrap(async (req, res) =>
+    res.send(await listInvoicesByLnurlPayEndpoint(req.params.id))))
+
+  // this is the actual endpoint users will hit
+  app.get('/lnurl/:id', wrap(async (req, res) => {
+    const lnurlpay = await getLnurlPayEndpoint(req.params.id)
+
+    res.status(200).send({
+      tag: 'payRequest'
+    , minSendable: lnurlpay.min
+    , maxSendable: lnurlpay.max
+    , metadata: makeMetadata(lnurlpay)
+    , commentAllowed: lnurlpay.comment
+    , callback: `https://${req.hostname}/lnurl/${lnurlpay.id}/callback`
+    })
+  }))
+
+  app.get('/lnurl/:id/callback', wrap(async (req, res) => {
+    const lnurlpay = await getLnurlPayEndpoint(req.params.id)
+
+    if (req.query.amount > lnurlpay.max)
```

Comment:
An amount that is `NaN` will pass both these tests, because `NaN` is not less-than or greater-than anything. For example, both `'foo' > 100` and  `'foo' < 200` return false.

I would start by explicitly casting `amount = +req.query.amount` first, then error if `!amount`, then continue to checking the min/max boundaries.